### ISSUE TITLE
VTM GO authenticitation is not working.

### ISSUE THREAD

**Issue by ldebt** [add-ons/plugin.video.vtm.go#394]
I made an installation on a new Google Chromecast device. When I open the VTM GO addon I have to make an authorisation via another device. Each time I get a notification "Login failed, check log file".
Installing VRT MAX was without any problem.

 - Google Chromecast 4K
 - Addon version: 1.4.4+matrix.1
 - Kodi version: V.  21.1
 - Operating System: Android

**Comment by keiem** [comment]
You should dowload the zip from the API-17 branch:

https://github.com/add-ons/plugin.video.vtm.go/tree/api-17

on the right, you see the tab "code". Click on it and download the zip.  Use this zip to install vtm-go.

**Comment by ldebt** [comment]
Solution works perfect on mediaplayer. Thank you very much therefore. However how can I solve the problem on my Google Chromecast, having no USB port ?

**Comment by ldebt** [comment]
Connection problem solved by using a powered usb-c hub. No further actions needed on this issue. Thanks

**Comment by jcjdammers** [comment]
Hi,

I am having the same authentication issue as above.
Addon version: 1.4.4+matrix.1
OS+Kodi: LibreELEC 12.0.1 with Kodi (Omega) v21.1

I have this addon installed from the official Kodi Addon repository. When do you push the fixed version to the Kodi repo?
Thanks.

**Comment by AlfaStation1** [comment]
Same problem: VTM GO authenticitation on my 2 different TV-boxes, my satelite decoder, and my Windows 10 pc have not worked for about 2months now.
Installing the API-17 zip solved the problem on the W10 machine and 1 of the boxes. 
I will try the rest later tonight but they run older Android systems so they might not meet the requirements anymore for dpg-media (VRT MAX still works on everything but VIER via Retro does not work anymore on the older boxes, neither does PLAY).

**Comment by jcjdammers** [comment]
Hi,
Where can I download the API-17 zip? The link above doesn't work for me.
Even better if the new version could be published to the Kodi Addon repo, so it updates automatically.
Any help is appreciated.

**Comment by davidbe** [comment]
You are right. I see that the branch was deleted two week ago: https://github.com/add-ons/plugin.video.vtm.go/pull/388 @michaelarnauts Have you more information?

**Comment by michaelarnauts** [comment]
I'm sorry but I have removed that branch.

Apparently, that branch could be used to play VTM Go programs without advertisements. This is something I have always attempted to happen correctly. I'm not sure what has happened, I guess it's due to a changed API version or maybe we are using a wrong API key.

DPG Media has contacted me about this, and since I don't have the time to make the necessary changes so the Add-on works on the latest API version and plays the advertisements as required, I have removed that branch.

I think the only changes needed are the ones that DPG requested in #389, and a bump of the API version to the latest version.

If anyone has the skills and free time to investigate this, please go ahead and create a Merge Request.

Maybe @mediaminister is interested in picking this up? He recently revived the GoPlay add-on. ❤️

I'm really sorry for not keeping up with the dedication I once had for maintaining the Add-ons. I started these Add-ons when there where not alternatives like the Android TV app we have today. I am not using Kodi anymore for quite a while.

Thanks everyone for the support over the years!

**Comment by jcjdammers** [comment]
@mediaminister , I would really appreciate if you could help here. Thanks.

**Comment by mediaminister** [comment]
I'm sorry, I _currently_ have no spare time to work on something I don't even watch.

Maybe another time...

**Comment by michaelarnauts** [comment]
> > Apparently, that branch could be used to play VTM Go programs without advertisements. This is something I have always attempted to happen correctly. I'm not sure what has happened, I guess it's due to a changed API version or maybe we are using a wrong API key.
> > DPG Media has contacted me about this, and since I don't have the time to make the necessary changes so the Add-on works on the latest API version and plays the advertisements as required, I have removed that branch.
> > I think the only changes needed are the ones that DPG requested in #389, and a bump of the API version to the latest version.
> 
> Doing what is requested in #389 is not enough. With the code in the deleted branch and this HEADER-adjustment, it not bring advertisements, as requested by DPG Media. To comply, a deeper dive is necessary.
> 

Probably, however I think there is a API key configured somewhere, and that has changed probably. I think this needs to be rolled back. DPG has made an API key for this add-on.

> > I'm really sorry for not keeping up with the dedication I once had for maintaining the Add-ons. I started these Add-ons when there where not alternatives like the Android TV app we have today. I am not using Kodi anymore for quite a while.
> 
> Just curious: what do you use now (instead of Kodi)?

The LG WebOS app for VTM and VRT, and I still have a Chromecast with Google TV hooked up for GoPlay. If only Play Media would release a LG WebOs app...

> > Thanks everyone for the support over the years!
> 
> Thank you for maintaining this add-on for all these years. Kodi & all those add-ons did make us NOT using cable for television - no need for me the cut the cable. ;-)

**Comment by AlfaStation1** [comment]
The API-17 version has also stopped working, meaning the VTM-GO app is now completely dead on Kodi.

If [michaelarnauts](https://github.com/michaelarnauts) is not using Kodi and/or does not have the time anymore, maybe [Basrieter ](https://github.com/retrospect-addon/plugin.video.retrospect/commits?author=basrieter) from the [Retrospect ](https://github.com/retrospect-addon/plugin.video.retrospect) team would be willing take over this project?


Many thanks to Michaelarnauts for maintaining the Vtm-go app all these years!

**Comment by davidbe** [comment]
The add-on is not completely dead... not yet.
In another issue, someone wrote what needs to be done **on the API-17-version**. A minor adjustment: in util.py "'x-app-version': '17'," should be version 18.

**Comment by AlfaStation1** [comment]
Thx [Davidbe](https://github.com/davidbe)! It worked 👍 
(now i can watch the Lieven Scheire AI show ;-) )

**Comment by jcjdammers** [comment]
@davidbe ,
Where can I download the API-17 version of the addon?
Thanks.

**Comment by AlfaStation1** [comment]
As i myself am unfamiliar to Github policies I might not be the right person to answer this but i'll give it a try:
The author (Michael) had to remove that branch due to DPGMedia having objections (that version allowed bypassing publicity ... yada-yada-yada ... as if people dont know how to record and skip adds anyways ... but i digress ;-) )
So I guess the only way is to pm people for it or just risk it by finding it on the web.
I believe the original file name was "plugin.video.vtm.go-api-17"

Feel free to delete this post if it goes against policy... but don't forget sites like Github exist because people like you and me believe the world is a better place when everyone shares and contributes freely to science and technology!

**Comment by davidbe** [comment]
@AlfaStation1 that seems to be correct, or better: that's how I think about it as well.
DPGMedia has been tolerating this add-on, as long it gives also the advertisement.

I want to invest some time to find out what has to be done so we can confirm to the demands of DPGMedia so this add-on can continue to exist. But my skills are somewhat basic.

**Comment by 8OND007** [comment]
any progress on a current working version ?

**Comment by PeterCloet** [comment]
> any progress on a current working version ?

Have a look here: https://github.com/add-ons/plugin.video.vtm.go/issues/405
